File: compiler/testData/loadJava/compiledJava/kotlinSignature/propagation/return/SubstitutedClassParameters.java
package test;

public interface SubstitutedClassParameters {

    public interface Super1<T> {
        T foo();

        void dummy(); // to avoid loading as SAM interface
    }

    public interface Super2<E> {
        E foo();

        void dummy(); // to avoid loading as SAM interface
    }

    public interface Sub extends Super1<String>, Super2<String> {
        String foo();
    }
}

File: kotlin-ide/idea/testData/compiler/loadJava/compiledKotlin/fromLoadJava/kotlinSignature/propagation/parameter/SubstitutedClassParameters.kt
package test

public interface SubstitutedClassParameters {

    public interface Super1<T> {
        public fun foo(t: T)

        public fun dummy() // to avoid loading as SAM interface
    }

    public interface Super2<E> {
        public fun foo(t: E)

        public fun dummy() // to avoid loading as SAM interface
    }

    public interface Sub: Super1<String>, Super2<String> {
        override fun foo(t: String)
    }
}
